xmppserver/src/main/java/org/jivesoftware/openfire/commands/admin/GetListRegisteredUsers.java

```diff
@@ -0,0 +1,136 @@
+
+        DataForm form = new DataForm(DataForm.Type.form);
+        form.setTitle(LocaleUtils.getLocalizedString("commands.admin.getlistregisteredusers.form.title", preferredLocale));
+        form.addInstruction(LocaleUtils.getLocalizedString("commands.admin.getlistregisteredusers.form.instruction", preferredLocale));
+
+        FormField field = form.addField();
+        field.setType(FormField.Type.hidden);
+        field.setVariable("FORM_TYPE");
+        field.addValue("http://jabber.org/protocol/admin");
+
+        field = form.addField();
+        field.setType(FormField.Type.list_single);
+        field.setLabel(LocaleUtils.getLocalizedString("commands.global.operation.pagination.max_items", preferredLocale));
+        field.setVariable("max_items");
+        field.addOption("25", "25");
+        field.addOption("50", "50");
+        field.addOption("75", "75");
+        field.addOption("100", "100");
+        field.addOption("150", "150");
+        field.addOption("200", "200");
+        field.addOption(LocaleUtils.getLocalizedString("commands.global.operation.pagination.none", preferredLocale), "none");
+
+        // Add the form to the command
+        command.add(form.getElement());
+    }
+
+    @Override
+    public void execute(@Nonnull final SessionData data, Element command) {
+        final Locale preferredLocale = SessionManager.getInstance().getLocaleForSession(data.getOwner());
+
+        int maxItems = -1;
+        final List<String> max_items_data = data.getData().get("max_items");
+        if (max_items_data != null && !max_items_data.isEmpty()) {
+            String max_items = max_items_data.get(0);
+            if (max_items != null && !"none".equals(max_items)) {
+                try {
+                    maxItems = Integer.parseInt(max_items);
+                } catch (NumberFormatException e) {
+                    // Do nothing. Assume that all users are being requested
+                }
+            }
+        }
+
+        DataForm form = new DataForm(DataForm.Type.result);
+
+        FormField field = form.addField();
+        field.setType(FormField.Type.hidden);
+        field.setVariable("FORM_TYPE");
+        field.addValue("http://jabber.org/protocol/admin");
+
+        field = form.addField();
+        field.setType(FormField.Type.jid_multi);
+        field.setLabel(LocaleUtils.getLocalizedString("commands.admin.getlistregisteredusers.form.field.registereduserjids.label", preferredLocale));
+        field.setVariable("registereduserjids");
+
+        // Get list of users (i.e. bareJIDs) that are connected to the server
+        final Collection<User> users = UserManager.getInstance().getUsers(0, maxItems);
+
+        // Add users to the result
+        for (User user : users) {
+            field.addValue(XMPPServer.getInstance().createJID(user.getUsername(), null));
+        }
+        command.add(form.getElement());
```

Comment:
This always appear to return 0 items

Reply:
Ah, Openfire logging from getUsers(...)

```
2024.01.13 16:58:07.359 ERROR [nioEventLoopGroup-3-10]: org.jivesoftware.database.DbConnectionManager - Disabling JDBC method pstmt.setFetchSize(fetchSize).
java.sql.SQLException: Invalid argument in JDBC call
    at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.7.1.jar:2.7.1]
    at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.7.1.jar:2.7.1]
    at org.hsqldb.jdbc.JDBCUtil.sqlException(Unknown Source) ~[hsqldb-2.7.1.jar:2.7.1]
    at org.hsqldb.jdbc.JDBCUtil.outOfRangeArgument(Unknown Source) ~[hsqldb-2.7.1.jar:2.7.1]
    at org.hsqldb.jdbc.JDBCPreparedStatement.setFetchSize(Unknown Source) ~[hsqldb-2.7.1.jar:2.7.1]
    at org.apache.commons.dbcp2.DelegatingStatement.setFetchSize(DelegatingStatement.java:742) ~[commons-dbcp2-2.9.0.jar:2.9.0]
    at org.apache.commons.dbcp2.DelegatingStatement.setFetchSize(DelegatingStatement.java:742) ~[commons-dbcp2-2.9.0.jar:2.9.0]
    at org.jivesoftware.database.DbConnectionManager.setFetchSize(DbConnectionManager.java:608) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.database.DbConnectionManager.limitRowsAndFetchSize(DbConnectionManager.java:589) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.user.DefaultUserProvider.getUsernames(DefaultUserProvider.java:263) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.user.DefaultUserProvider.getUsers(DefaultUserProvider.java:288) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.user.UserManager.getUsers(UserManager.java:349) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.commands.admin.GetListRegisteredUsers.execute(GetListRegisteredUsers.java:103) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.commands.AdHocCommandManager.process(AdHocCommandManager.java:286) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.commands.AdHocCommandHandler.handleIQ(AdHocCommandHandler.java:80) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.handler.IQHandler.process(IQHandler.java:102) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.IQRouter.handle(IQRouter.java:403) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.IQRouter.route(IQRouter.java:106) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.spi.PacketRouterImpl.route(PacketRouterImpl.java:74) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.net.StanzaHandler.processIQ(StanzaHandler.java:392) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.net.ClientStanzaHandler.processIQ(ClientStanzaHandler.java:90) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.net.StanzaHandler.process(StanzaHandler.java:334) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.net.StanzaHandler.processStanza(StanzaHandler.java:222) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.net.StanzaHandler.process(StanzaHandler.java:114) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.nio.NettyConnectionHandler.channelRead0(NettyConnectionHandler.java:142) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at org.jivesoftware.openfire.nio.NettyConnectionHandler.channelRead0(NettyConnectionHandler.java:50) [xmppserver-4.8.0-SNAPSHOT.jar:4.8.0-SNAPSHOT]
    at io.netty.channel.SimpleChannelInboundHandler.channelRead(SimpleChannelInboundHandler.java:99) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.handler.timeout.IdleStateHandler.channelRead(IdleStateHandler.java:286) [netty-handler-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.handler.codec.ByteToMessageDecoder.fireChannelRead(ByteToMessageDecoder.java:346) [netty-codec-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.handler.codec.ByteToMessageDecoder.channelRead(ByteToMessageDecoder.java:318) [netty-codec-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:444) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.handler.traffic.AbstractTrafficShapingHandler.channelRead(AbstractTrafficShapingHandler.java:506) [netty-handler-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:442) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.fireChannelRead(AbstractChannelHandlerContext.java:412) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.DefaultChannelPipeline$HeadContext.channelRead(DefaultChannelPipeline.java:1410) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:440) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.AbstractChannelHandlerContext.invokeChannelRead(AbstractChannelHandlerContext.java:420) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.DefaultChannelPipeline.fireChannelRead(DefaultChannelPipeline.java:919) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.nio.AbstractNioByteChannel$NioByteUnsafe.read(AbstractNioByteChannel.java:166) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:788) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:724) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:650) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:562) [netty-transport-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.util.concurrent.SingleThreadEventExecutor$4.run(SingleThreadEventExecutor.java:997) [netty-common-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.util.internal.ThreadExecutorMap$2.run(ThreadExecutorMap.java:74) [netty-common-4.1.100.Final.jar:4.1.100.Final]
    at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30) [netty-common-4.1.100.Final.jar:4.1.100.Final]
    at java.lang.Thread.run(Unknown Source) [?:?]
Caused by: org.hsqldb.HsqlException: Invalid argument in JDBC call
    at org.hsqldb.error.Error.error(Unknown Source) ~[hsqldb-2.7.1.jar:2.7.1]
```

Reply:
Hmm, I don't think this error would cause the database query to fail. The log level 'error' is somewhat dramatic - it basically tries to optimize a query, fails, and sets a flag so that it won't try to do that again in the future.

Reply:
I have now pushed a commit that reduces the log level of these messages.